packages/@sanity/desk-tool/src/components/confirmDeleteDialog/useReferringDocuments.ts

```diff
@@ -92,16 +101,27 @@ function fetchCrossDatasetReferences(
     switchMap(() => fetchCrossDatasetTokens()),
     switchMap((crossDatasetTokens) => {
       const currentDataset = client.config().dataset
-      return versionedClient.observable.request({
-        url: `/data/references/${currentDataset}/documents/${documentId}/to?excludeInternalReferences=true&excludePaths=true`,
-        ...(crossDatasetTokens.length > 0
-          ? {
-              'sanity-project-tokens': crossDatasetTokens
-                .map((t) => `${t.projectId}=${t.token}`)
-                .join(','),
-            }
-          : null),
-      })
+
+      return versionedClient.observable
+        .request({
+          url: `/data/references/${currentDataset}/documents/${documentId}/to?excludeInternalReferences=true&excludePaths=true`,
+          ...(crossDatasetTokens.length > 0
+            ? {
+                'sanity-project-tokens': crossDatasetTokens
+                  .map((t) => `${t.projectId}=${t.token}`)
+                  .join(','),
+              }
+            : null),
+        })
+        .pipe(
+          catchError((e) => {
+            // it's possible that referencing document doesn't exist yet so the
+            // API will return a 404. In those cases, we want to catch and return
+            // a response with no references
+            if (isClientError(e)) return of({totalCount: 0, references: []})
+            throw e
```

Comment:
This is the way to rethrow

Alternatively you could import `throwError` from `rxjs` and return `throwError(e)`, but it should be pretty much equivalent (the stack trace will be preserved in both cases afaik), so I'd prefer `throw`